core/storage/activity/gae_models.py

```diff
@@ -37,25 +43,33 @@ class ActivityReferencesModel(base_models.BaseModel):
     # in this list is a dict with two keys: 'type' and 'id'.
     activity_references = datastore_services.JsonProperty(repeated=True)
 
+    def __init__(self, *args, **kwargs): # pylint: disable=useless-super-delegation
```

Comment:
Here and in other files, why do you need to add this?

Reply:
@seanlip I added __init__ in models where they are not present for mypy to work correctly. If the constructor is not added, MyPy assumes the subclass variable's object to be that of Parent class.
Wrote this in the description also. PTAL. Thanks!

Reply:
I haven't seen any proof that this is unavoidable or that this is the only way to do it. 

It also seems a bit odd that MyPy does that. Can you point to any documentation around this issue? Have you asked on the MyPy community channels whether this is an error with MyPy? I don't see, from a conceptual perspective, why the constructor is needed here.

Reply:
I encounter it and asked the community about it. Haven't received a reply till now. 
If I didn't add constructor to VersionedModel(which is subclas of BaseModel) then its instances using VersionedModel() were thought by mypy to be of type BaseModel

Reply:
Perhaps try asking again? See if you can point to the discussion here too, so that others can follow along. Have you filed an issue?

Also, is the issue actually a generic one due to MyPy or is it specific to ndb.Model? Try creating a minimal new project, with something like:

```
class Base(object):
    def __init__(self):
         self.x = 42

class Derived(Base):
    pass

def f(x: Derived) -> None:
    return 10

derived_instance = Derived()
print f(derived_instance)
```

and then run MyPy on it. Does MyPy throw an error?

Reply:
@seanlip Tried this. This error occurs only when we subclass something of type `Any`.
```
# x is of type Any

class A(x):
    def __init__(self, *args, **kwargs):
        # type: (*Any, **Any) -> None
        self.a = 1

class B(A):
    pass

class C(B):
    pass

a = A()
b = B()
c = C()

reveal_type(a)
reveal_type(b)
reveal_type(c)
```
This outputs:
```
../../t.py:19: note: Revealed type is 't.A'
../../t.py:20: note: Revealed type is 't.A'
../../t.py:21: note: Revealed type is 't.A'
```

If I use `object` in place of `x`, the parent class is not `Any`. Now the output is:
```
../../t.py:19: note: Revealed type is 't.A'
../../t.py:20: note: Revealed type is 't.B'
../../t.py:21: note: Revealed type is 't.C'
``` 

I will report this bug. For now using __init__ seems a good workaround right?

Reply:
Nope (but thanks for investigating). Let's try and find a way to get the parent to not be Any. Fixing this issue now will make a lot of problems later go away.

Reply:
@seanlip Thanks! How should I proceed with this? I'm clueless about this. 
Just asking, why adding inits is not a good idea?

Reply:
This, you will need to figure out.

Adding code that necessitates our use of pylint-disables is typically not a good idea.

Reply:
Added stubs